test/index.js

```diff
@@ -209,6 +211,27 @@ testCommand ('bin/doctest --module commonjs lib/doctest.js', {
   stderr: ''
 });
 
+if (esmSupported) {
+  testCommand ('bin/doctest --module esm test/esm/index.mjs', {
```

Comment:
Unless we add a special `.mjs` test file, along with all the new tooling needed, this functionality can only be tested through `testCommand`.